string msg_type
string domain
string variable
string value
string note
